lib/rex/text.rb

```diff
@@ -1522,7 +1522,8 @@ def self.badchar_index(data, badchars = '')
   # @param data [#delete]
   # @param badchars [String] A list of characters considered to be bad
   def self.remove_badchars(data, badchars = '')
-    data.delete(badchars)
+    badchars_pat = badchars.unpack("C*").map{|c| "\\x%.2x" % c}.join
+    data.gsub!(/[#{badchars_pat}]/n, '')
```

Comment:
`String#gsub!` will return `nil` if it didn't substitute anything which will be different behavior from what happened before with `#delete`.

`String#gsub` is safer